packages/core/src/core/client.ts

```diff
@@ -410,8 +410,9 @@ export class GeminiClient {
     contents: Content[],
     generationConfig: GenerateContentConfig,
     abortSignal: AbortSignal,
+    model?: string,
   ): Promise<GenerateContentResponse> {
-    const modelToUse = this.config.getModel();
+    const modelToUse = model ?? this.config.getModel();
```

Comment:
Why the change to the selected model here? This seems maybe unrelated?

Reply:
I want to override the mode if specified by the config when calling this function.